code/game/objects/items/robot/items/hypo.dm

```diff
@@ -370,6 +374,69 @@
 		shaker.trans_to(target, amount_per_transfer_from_this, transfered_by = user)
 		balloon_alert(user, "[amount_per_transfer_from_this] unit\s poured")
 	return .
+	
+/obj/item/reagent_containers/borghypo/condimentsynthesizer // Solids! Condiments! The borger uprising!
+	name = "Condiment Synthesizer"
+	desc = "An advanced condiment synthesizer"
+	icon = 'icons/obj/food/containers.dmi'
+	icon_state = "flour"
+	possible_transfer_amounts = list(5,10,20)
+	// Lots of reagents all regenerating at once, so the charge cost is lower. They also regenerate faster.
+	charge_cost = 40 //Costs double the power of the borgshaker due to synthesizing solids
+	recharge_time = 6 //Double the recharge time too, for the same reason.
+	dispensed_temperature = WATER_MATTERSTATE_CHANGE_TEMP 
+	default_reagent_types = EXPANDED_SERVICE_REAGENTS
+
+/obj/item/reagent_containers/borghypo/condimentsynthesizer/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "BorgHypo", name)
+		ui.open()
+
+/obj/item/reagent_containers/borghypo/condimentsynthesizer/ui_data(mob/user)
+	var/list/condiments = list()
+	for(var/datum/reagent/reagent in stored_reagents.reagent_list)
+		if(reagent)
+			condiments.Add(list(list(
+				"name" = reagent.name,
+				"volume" = round(reagent.volume, 0.01) - 1,
+			))) // list in a list because Byond merges the first list...
+
+	var/data = list()
+	data["theme"] = tgui_theme
+	data["minVolume"] = amount_per_transfer_from_this
+	data["maxVolume"] = max_volume_per_reagent
+	data["reagents"] = condiments
+	data["selectedReagent"] = selected_reagent?.name
+	return data
+		
+/obj/item/reagent_containers/borghypo/condimentbag/attack(mob/M, mob/user)
+	return //Can't inject stuff with a shaker, can we? //not with that attitude
+
+/obj/item/reagent_containers/borghypo/condimentsynthesizer/afterattack(obj/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return .
+	if(!selected_reagent)
+		balloon_alert(user, "no reagent selected!")
+		return .
+	. |= AFTERATTACK_PROCESSED_ITEM
+	if(target.is_refillable())
```

Comment:
can we turn this into an early return like `if(!target.is_refillable())` then `return .`? would save a bit of ugly indentation.